### DISCUSSION TITLE
Rocketmq 5.x版本的消息发送的同步问题。

### DISCUSSION THREAD

**Discussion by AI1186780944** [apache/rocketmq#7107] [Q&A]
服务器版本：rocketmq：5.1.3，
客户端版本：rocketmq-client-java：5.0.5，
服务运行模式：Local模式，
现象描述：
1、在使用同步发送消息的情况下，消费者不处理完消息，生产者没办法在send()方法中获取到MessageId的结果值。
2、基于第1条现象，在关闭消费者的情况下，消息又能正常的发送，并且无视消费者的存在与否。
问题：
请介绍一下这个消息推送的原理，在新版本中消息到底是如何处理的？生产者同步发送消息，是否必须等待消费者消费完消息。能否按照旧版本的机制，理解成生产者发送完消息以后，不关心消费者的消费情况，只根据发送结果来判定消息的发送状态。

**Comment by ShadowySpirits** [comment]
同步发送消息的情况下，消息存储在 broker 上即返回给生产者响应，不会等待消费者消费
